### PR TITLE
[Docs] Flesh out the rules for concept-exercies.md slightly more

### PR DESCRIPTION
**Pull request by iHiD**
We're suffering from people not reading these docs. I'm putting a video to help. This PR addresses the following to help:

- Clearly specifies the purpose of each file to give quick overviews
- More bullet points and less prose to provide extra information
- Incorporates some of @mikedamay's points in https://github.com/exercism/v3/issues/1087

We can improve the other sections later - but if everyone agrees this is an improvement, let's get it down on master.

### CODE REVIEW

File: docs/concept-exercises.md [hunk 1]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+**Purpose:** Introduce the concept(s) that the exercise teaches to the student.
 
-As an example, the introduction to a "strings" exercise might describe a string as just a "Sequence of Unicode characters" or a "series of bytes". Unless the student needs to understand more nuanced details in order to solve the exercise, this type of brief explanation (along with an example of its syntax) should be sufficient information for the student to solve the exercise.
+- The information provided should give the student just enough context to figure out the solution themselves.
```
Comment (SaschaMann) [inline docs/concept-exercises.md:56]: If it provides them with a link that contains required information, is that also valid? Or should we assume their internet isn't working and that they should be able to solve the exercise just with the information in this file?

File: docs/concept-exercises.md [hunk 2]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 
 See the C# floating-point-numbers exercise's [introduction.md file][csharp-docs-introduction.md] for an example.
 
 ### `.docs/instructions.md`
 
-This file contains instructions for the exercise. It should explicitly explain what the student needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+**Purpose:** Provide instructions for the exercise.
+
+This file is split into two halves. 
+1. The first half explains the "story" or "theme" of the exericise. It should generally contain no code samples.
```
Comment (SaschaMann) [inline docs/concept-exercises.md:70]:
```suggestion
1. The first section explains the "story" or "theme" of the exercise. It should generally contain no code samples.
```

The narrative might not take up half the instructions. Unless we want to add a length requirement/suggestion here, I think "part" or "section" work better. What do you think?

---

Just the typo:

```suggestion
1. The first half explains the "story" or "theme" of the exercise. It should generally contain no code samples.
```

File: docs/concept-exercises.md [hunk 3]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+2. The second half provides clear instructions of what a student needs to do, providing tasks.
```
Comment (SaschaMann) [inline docs/concept-exercises.md:71]:
```suggestion
2. The second section provides clear instructions of what a student needs to do, providing tasks.
```

File: docs/concept-exercises.md [hunk 4]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+
+Each tasks must conform to the following standard:
+- Start with a third-level heading starting with a number (.e.g `### 1.`, `### 2.`)
```
Comment (SaschaMann) [inline docs/concept-exercises.md:74]: Does the narrative part have a heading? Could we check this automatically?

File: docs/concept-exercises.md [hunk 5]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+- Provide a method that a student needs to define with its signature(e.g. `X(...)` that takes an A and returns a Z),
```
Comment (SaschaMann) [inline docs/concept-exercises.md:75]:
```suggestion
- Provide a method that a student needs to define with its signature (e.g. `X(...)` that takes an A and returns a Z),
```

Shouldn't this go into the exercise stub file?

This is track specific and I'm not sure if it should be considered in this file but on the Julia track we don't necessarily want to provide (type) signatures because it's often not a good practice to write them out. Would it be sufficient to provide something like "define method `foo` that takes two arguments" instead?

File: docs/concept-exercises.md [hunk 6]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+- Provide at example usage of that function. These examples should be different examples to those given in the tests.
```
Comment (SaschaMann) [inline docs/concept-exercises.md:76]:
```suggestion
- Provide an example usage of that function. These examples should be different examples to those given in the tests.
```

File: docs/concept-exercises.md [hunk 7]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 
-See the C# floating-point-numbers exercise's [instructions.md file][csharp-docs-instructions.md] for an example.
+Avoid any controversial topics in the stories - be empathetic to triggers that people might have - there's normally a less controversial or risque variant of a story that can be used.
```
Comment (SaschaMann) [inline docs/concept-exercises.md:78]: Could you define "controversial"?

File: docs/concept-exercises.md [hunk 8]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+
+See the C# numbers exercise's [instructions.md file][csharp-docs-instructions.md] for an example. Notice the clear distinction between the story at the top and the tasks with code samples below.
 
 ### `.docs/hints.md`
 
-If the student gets stuck, we will allow them to click a button requesting a hint, which shows this file. This will not be a "recommended" path and we will (softly) discourage them using it unless they can't progress without it. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated.
+**Purpose:** Provide hints to a student to help them get themselves unstuck n an exercise.
```
Comment (SaschaMann) [inline docs/concept-exercises.md:84]:
```suggestion
**Purpose:** Provide hints to a student to help them get themselves unstuck in an exercise.
```

File: docs/concept-exercises.md [hunk 9]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 
-The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any student. They might link to the docs of the functions that need to be used.
+- If the student gets stuck, we will allow them to click a button requesting a hint, which will show the relevant part of file. 
+- Hints should be bullet-pointed underneath headings.
+- General hints about the exericse can appear under the`### General` heading.
```
Comment (SaschaMann) [inline docs/concept-exercises.md:88]:
```suggestion
- General hints about the exericse can appear under the `### General` heading.
```

File: docs/concept-exercises.md [hunk 10]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+- Task-specific hints should appear undernearth headings that match their task in the `instructions.md` (e.g. `### 2.`).
```
Comment (SaschaMann) [inline docs/concept-exercises.md:89]:
```suggestion
- Task-specific hints should appear underneath headings that match their task in the `instructions.md` (e.g. `### 2.`).
```

File: docs/concept-exercises.md [hunk 11]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 
-The hints should not spell out the solution, but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
+Viewing hints will not be a "recommended" path and we will (softly) discourage them using it unless they can't progress without it. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated. These hints should be enough to unblock almost any student. They might link to the docs of the functions that need to be used. The hints should not spell out the solution, but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
```
Comment (SaschaMann) [inline docs/concept-exercises.md:91]:
```suggestion
Viewing hints will not be a "recommended" path and we will (softly) discourage using it unless the student can't progress without it. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated. These hints should be enough to unblock almost any student. They might link to the docs of the functions that need to be used. The hints should not spell out the solution, but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
```

File: docs/concept-exercises.md (hunk 4 above)
Comment (iHiD) [reply docs/concept-exercises.md:74]: No it doesn't, and it shouldn't. We could potentially check this if you/Erik fancied adding some CI for this and the other headings?

File: docs/concept-exercises.md (hunk 1 above)
Comment (iHiD) [reply docs/concept-exercises.md:56]: My issue with links in the introduction is that the page linked to will almost always contain loads of *extra* information that overloads the student. If we could avoid that, then I'd be ok with links, and for complex topics (e.g. recursion) there might be great resources to link to. But generally I think explaining things concisely in the introduction is the right thing to do.

Comment (SaschaMann) [reply docs/concept-exercises.md:56]: I agree with that. Could you perhaps add the info from your comment explicitely to the doc?

File: docs/concept-exercises.md (hunk 4 above)
Comment (SaschaMann) [reply docs/concept-exercises.md:74]: > We could potentially check this if you/Erik fancied adding some CI for this and the other headings?

Should be doable. I dislike paying attention to conform to formatting guidelines manually, so I think having a check makes it easier. I could add a check that each heading is a level 3 heading and starts with an increasing counter, any cases I'm missing? Are subheadings in the tasks allowed?

File: docs/concept-exercises.md (hunk 5 above)
Comment (wolf99) [reply docs/concept-exercises.md:75]: Additionally, I could see this might get in the way of having useful exercises for some concepts, e.g. functions, which might require a student to parse such a requirement from prose and implement it

File: docs/concept-exercises.md (hunk 7 above)
Comment (efx) [reply docs/concept-exercises.md:78]: I am also curious too but don't think we need to delay merging the PR on a definition. I would interpret this by Exercism's [code of conduct](https://exercism.io/code-of-conduct).

File: docs/concept-exercises.md [hunk 12]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 ### `.docs/instructions.md`
 
-This file contains instructions for the exercise. It should explicitly explain what the student needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+**Purpose:** Provide instructions for the exercise.
+
+This file is split into two halves. 
+1. The first half explains the "story" or "theme" of the exercise. It should generally contain no code samples.
+2. The second half provides clear instructions of what a student needs to do, providing tasks.
+
+Each tasks must conform to the following standard:
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md:73]:
```suggestion
Each task must conform to the following standard:
```

File: docs/concept-exercises.md (hunk 5 above)
Comment (ErikSchierboom) [reply docs/concept-exercises.md:75]: I didn't immediately understand what this rule wanted me to do. Maybe rephrase to something like:

```suggestion
- Describe which function/method the student needs to define/implement (e.g. `Implement method X(...)` that takes an A and returns a Z),
```

File: docs/concept-exercises.md [hunk 13]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 
 ### `.docs/hints.md`
 
-If the student gets stuck, we will allow them to click a button requesting a hint, which shows this file. This will not be a "recommended" path and we will (softly) discourage them using it unless they can't progress without it. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated.
+**Purpose:** Provide hints to a student to help them get themselves unstuck in an exercise.
 
-The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any student. They might link to the docs of the functions that need to be used.
+- If the student gets stuck, we will allow them to click a button requesting a hint, which will show the relevant part of file. 
+- Hints should be bullet-pointed underneath headings.
+- General hints about the exericse can appear under the `### General` heading.
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md:88]:
```suggestion
- General hints about the exercise can appear under the `### General` heading.
```

File: docs/concept-exercises.md [hunk 14]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+- Task-specific hints should appear underneath headings that match their task in the `instructions.md` (e.g. `### 2.`).
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md:89]:
```suggestion
- Task-specific hints should appear underneath headings that match their task heading in the `instructions.md` (e.g. `### 2.`).
```

File: docs/concept-exercises.md [hunk 15]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 
-The hints should not spell out the solution, but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
+Viewing hints will not be a "recommended" path and we will (softly) discourage them using it unless they can't progress without it. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated. These hints should be enough to unblock almost any student. They might link to the docs of the functions that need to be used. The hints should not spell out the solution, but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
```
Comment (ErikSchierboom) [inline docs/concept-exercises.md:91]:
```suggestion
Viewing hints will not be a "recommended" path and we will (softly) discourage students using them unless they can't progress without them. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated. These hints should be enough to unblock almost any student. They might link to the docs of the functions that need to be used. The hints should not spell out the solution (no code examples or method/function names), but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
```

File: docs/concept-exercises.md [hunk 16]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+**Purpose:** Introduce the concept(s) that the exercise teaches to the student.
 
-As an example, the introduction to a "strings" exercise might describe a string as just a "Sequence of Unicode characters" or a "series of bytes". Unless the student needs to understand more nuanced details in order to solve the exercise, this type of brief explanation (along with an example of its syntax) should be sufficient information for the student to solve the exercise.
+- The information provided should give the student just enough context to figure out the solution themselves. 
+- It should not give the student any extra information about a concept not needed to understand the fundamentals of the concept or solve the exercise.
```
[the remark this answers is not in the document]
Comment (iHiD) [reply docs/concept-exercises.md:57]: I'd rather this was about the file, not the author.

```suggestion
- The file should not give the student any extra information about a concept not needed to understand the fundamentals of the concept or solve the exercise.
```

File: docs/concept-exercises.md (hunk 3 above)
Comment (iHiD) [reply docs/concept-exercises.md:71]:
```suggestion
2. The second part provides clear instructions of what a student needs to do, providing tasks.
```

File: docs/concept-exercises.md [hunk 17]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
 
 See the C# floating-point-numbers exercise's [introduction.md file][csharp-docs-introduction.md] for an example.
 
 ### `.docs/instructions.md`
 
-This file contains instructions for the exercise. It should explicitly explain what the student needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+**Purpose:** Provide instructions for the exercise.
+
+This file is split into two halves. 
+1. The first half explains the "story" or "theme" of the exercise. It should generally contain no code samples.
```
Comment (iHiD) [inline docs/concept-exercises.md:70]:
```suggestion
1. The first part explains the "story" or "theme" of the exercise. It should generally contain no code samples.
```

File: docs/concept-exercises.md [hunk 18]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+As an example, the introduction to a "strings" exercise might describe a string as just a "Sequence of Unicode characters" or a "series of bytes", tell the users how to create a string, and explain that a string has methods that can be used to manipulate it. Unless the student needs to understand more nuanced details in order to solve the exercise, this type of brief explanation (along with an example of its syntax) should be sufficient information for the student to solve the exercise.
 
 See the C# floating-point-numbers exercise's [introduction.md file][csharp-docs-introduction.md] for an example.
 
 ### `.docs/instructions.md`
 
-This file contains instructions for the exercise. It should explicitly explain what the student needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+**Purpose:** Provide instructions for the exercise.
+
+This file is split into two halves.
```
Comment (iHiD) [inline docs/concept-exercises.md:69]:
```suggestion
This file is split into two parts. 
```

File: docs/concept-exercises.md (hunk 5 above)
Comment (iHiD) [reply docs/concept-exercises.md:75]: > Additionally, I could see this might get in the way of having useful exercises for some concepts, e.g. functions, which might require a student to parse such a requirement from prose and implement it. (i.e. writing the signature is the exercise)

@wolf99 Good point and I agree, but I'm going to leave this out of here, as it's a one-off exception to the rule, which I hope maintainers will be able to extrapolate.

File: docs/concept-exercises.md [hunk 19]
```diff
@@ -51,30 +51,51 @@ An exercise has the following files. In the browser they will show at the releva
...
+1. The first part explains the "story" or "theme" of the exercise. It should generally contain no code samples.
+2. The second part provides clear instructions of what a student needs to do, providing tasks.
+
+Each task must conform to the following standard:
+- Start with a third-level heading starting with a number (.e.g `### 1.`, `### 2.`)
+- Describe which function/method the student needs to define/implement (e.g. `Implement method X(...)` that takes an A and returns a Z),
+- Provide an example usage of that function. These examples should be different examples to those given in the tests.
 
-See the C# floating-point-numbers exercise's [instructions.md file][csharp-docs-instructions.md] for an example.
+Avoid any controversial topics in the stories - be empathetic to triggers that people might have - there's normally a less controversial or risque variant of a story that can be used.
```
Comment (iHiD) [inline docs/concept-exercises.md:78]:
```suggestion
Avoid any topics in the stories that could be misconstrued or cause offence - be empathetic to triggers that people might have - there's normally a less controversial or risque variant of a story that can be used.
```

File: docs/concept-exercises.md (hunk 7 above)
Comment (iHiD) [reply docs/concept-exercises.md:78]: Anything that might offend anyone. In one example exercise there was a casual reference to Covid-19, which while not in itself malicious or nasty, could easily be triggering by someone who lost a friend to it. I've clarified in a suggestion below.

File: docs/concept-exercises.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply docs/concept-exercises.md:57]: That works for me.

File: docs/concept-exercises.md (hunk 7 above)
Comment (SaschaMann) [reply docs/concept-exercises.md]: > but don't think we need to delay merging the PR on a definition

I think it should either be left out of this PR or clarified. The CoC applies anyway and this goes beyond it and has the potential to violate the spirit of the CoC itself.

If you're not a native speaker (or in many cases not directly familiar with US-American culture), it can be incredibly hard to figure out what you're "allowed" to say or not. There has been one recent case in problem-specs where someone wanted to change phrasing of some examples because they considered it too offensive or vulgar even though it wasn't a violation of the CoC. Having such a vague requirement in this doc can scare people away from writing exercises in the first place, especially when there have been examples where something was "controversial" beyond the way the CoC specifies it. Scaring people away from writing exercises is generally not something we want to achieve, I think.

Instead of a negative requirement, some encouragement would be better if you want to leave an additional hint in this doc. Something like "In the spirit of the CoC, we want to keep exercise stories welcoming to everyone, here are some things you could include to ensure that: <list of suggestions>" or "Please keep the CoC in mind when writing the story for your exercise".

File: docs/concept-exercises.md (hunk 4 above)
Comment (iHiD) [reply docs/concept-exercises.md:74]: I'm going to resolve this one, but I've asked @ErikSchierboom to create an issue to remind us to add this to the CI.

File: docs/concept-exercises.md (hunk 1 above)
Comment (iHiD) [inline docs/concept-exercises.md:56]:
```suggestion
- The information provided should give the student just enough context to figure out the solution themselves. 
- Links should be used sparingly if at all. While a link explaining a complex topic like recursion might be useful, for most concepts the links will provide more information than needed so explaining things concisely inline should be the aim.
```

File: docs/concept-exercises.md (hunk 19 above)
Comment (SaschaMann) [reply docs/concept-exercises.md]: This still has the same issues that I outlined in my comment above. I wrote it at the same time you added this suggestion :D

Comment (SleeplessByte) [reply docs/concept-exercises.md]: Perhaps instead we can turn this into a _positive_? 

Instead of **avoid x y z** which requires interpretation of the writer, we could do something like **some topics might be misconstrued or cause offence. When in doubt if this will be the case, ping @exercism/xyz in your PR**

File: docs/concept-exercises.md [hunk 20]
```diff
@@ -51,30 +51,52 @@ An exercise has the following files. In the browser they will show at the releva
...
+1. The first part explains the "story" or "theme" of the exercise. It should generally contain no code samples.
+2. The second part provides clear instructions of what a student needs to do, providing tasks.
+
+Each task must conform to the following standard:
+- Start with a third-level heading starting with a number (.e.g `### 1.`, `### 2.`)
+- Describe which function/method the student needs to define/implement (e.g. `Implement method X(...)` that takes an A and returns a Z),
+- Provide an example usage of that function. These examples should be different examples to those given in the tests.
 
-See the C# floating-point-numbers exercise's [instructions.md file][csharp-docs-instructions.md] for an example.
+Avoid any topics in the stories that could be misconstrued or cause offence - be empathetic to triggers that people might have - there's normally a less controversial or risque variant of a story that can be used.
```
Comment (iHiD) [inline docs/concept-exercises.md:79]:
```suggestion
Try and avoid any topics in the stories that could be misconstrued or cause offence and be empathetic to triggers that people might have. If you are unsure if something might work or not, please mention @exercism/leadership and we can clarify. Please don't be disappointed or frustrated if we ever request a story to change due to the risk of offending. We place high value on making the content safe for everyone and so often err on the side of caution.
```

Comment (iHiD) [reply docs/concept-exercises.md:79]: @SaschaMann @SleeplessByte @ErikSchierboom?

Comment (iHiD) [inline docs/concept-exercises.md:79]:
```suggestion
```
